demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +141,316 @@ def __init__(
-                "format is not supported when the DEMISTO_SDK_OFFLINE_ENV environment variable is set, skipping it."
+                "Formatting is not supported when the DEMISTO_SDK_OFFLINE_ENV environment variable is set, Skipping..."
             )
 
     def download(self) -> int:
         """
-        Downloads custom content data from Demisto to the output pack in content repository.
-        :return: The exit code
+        Downloads custom content data from XSOAR to the output path provided.
+
+        Returns:
+            int: Exit code. 1 if failed, 0 if succeeded
         """
-        exit_code: int = self.download_manager()
-        self.remove_traces()
-        return exit_code
+        return self.download_manager()
 
     def download_manager(self) -> int:
         """
         Manages all download command flows
-        :return The exit code of each flow
+
+        Returns:
+            int: Exit code. 1 if failed, 0 if succeeded
         """
-        if not self.verify_flags():
-            return 1
-        if not self.download_system_item and not self.fetch_custom_content():
-            return 1
-        if self.download_system_item and not self.fetch_system_content():
-            return 1
-        if self.handle_list_files_flag():
-            return 0
-        self.handle_init_flag()
-        self.handle_all_custom_content_flag()
-        self.handle_regex_flag()
-        if not self.verify_output_pack_is_pack():
-            return 1
+        try:
+            if not (self.verify_flags() and self.verify_output_path()):
+                return 1
 
-        self.build_pack_content()
-        self.build_custom_content() if not self.download_system_item else self.build_system_content()
-        self.update_pack_hierarchy()
-        self.merge_into_pack()
-        self.log_files_downloaded()
-        self.log_files_not_downloaded()
+            if self.init:
+                self.initialize_output_path()
 
-        for entry in self.files_not_downloaded:
-            file, reason = entry
-            if reason != FILE_EXIST_REASON:
-                return 1
+            if self.download_system_items and not self.list_files:
+                downloaded_content_objects = self.fetch_system_content()
+
+            else:
+                all_custom_content_data = self.download_custom_content()
+                all_custom_content_objects = self.parse_custom_content_data(custom_content_data=all_custom_content_data)
+
+                # If we're in list-files mode, print the list of all available custom content and exit
+                if self.list_files:
+                    logger.info(f"list-files (-lf) mode detected. Listing available custom content files "
+                                f"({len(all_custom_content_objects)}):")
+                    table_str = self.create_custom_content_table(custom_content_objects=all_custom_content_objects)
+
+                    logger.info(table_str)
+                    return 0
```

Comment:
add a comment explaining why we return mid-function

Reply:
Line 192 (above this `if` scope):
```python
 # If we're in list-files mode, print the list of all available custom content and exit
```